### PR TITLE
MOD-6198 add vecsim info to ft.info

### PR DESCRIPTION
**Pull request by meiravgri**
This PR enhances the ft.info command by adding vector field information.
**For both HNSW and FLAT** algorithms the attributes include 
* data_type
* dim
* distance_metric.

**For HNSW**, additional attributes are also displayed:
* M
* ef_construction 

output examples:

**HNSW:**
```
 8) 1)  1) identifier
        2) vec2
        3) attribute
        4) vec2
        5) type
        6) VECTOR
        7) algorithm
        8) HNSW
        9) data_type
       10) FLOAT32
       11) dim
       12) (integer) 128
       13) distance_metric
       14) L2
       15) M
       16) (integer) 16
       17) ef_construction
       18) (integer) 200
```

**FLAT:**
``` 8) 1)  1) identifier
        2) vec
        3) attribute
        4) vec
        5) type
        6) VECTOR
        7) algorithm
        8) FLAT
        9) data_type
       10) FLOAT32
       11) dim
       12) (integer) 128
       13) distance_metric
       14) L2
```

### CODE REVIEW

File: tests/pytests/test_info.py [hunk 1]
```diff
@@ -52,6 +52,37 @@ def testInfo(env):
...
   #print info
 
+def test_vecsim_hnsw_info():
+  env = Env(protocol=3)
+  dim = 2
+
+  for alg in ["HNSW", "FLAT"]:
+    info_expected = {"identifier": "vec", "attribute": "vec", "type": "VECTOR", "algorithm": alg,
+                     "dim": dim, "flags": []}
+    hnsw_params = {"M": 12, "ef_construction": 100} if alg == "HNSW" else {}
```
Comment (alonre24) [inline tests/pytests/test_info.py:62]:
```suggestion
    additional_params = {"M": 12, "ef_construction": 100} if alg == "HNSW" else {}
```
would be clearer IMO

File: tests/pytests/test_info.py [hunk 2]
```diff
@@ -52,6 +52,37 @@ def testInfo(env):
...
+      for metric in ["L2", "IP", "COSINE"]:
+        info_expected["distance_metric"] = metric
+        # create index
+        params = ["TYPE", type, "DIM", dim,
+                  "DISTANCE_METRIC", metric, *to_list(hnsw_params)]
+
+        env.expect('FT.CREATE', "idx",
+                   'SCHEMA', "vec", 'VECTOR',
+                   alg, len(params),
+                   *params).ok()
```
Comment (alonre24) [inline tests/pytests/test_info.py:77]: Why so many lines?

File: src/info_command.c [hunk 3]
```diff
@@ -157,6 +157,29 @@ int IndexInfoCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
       geom_idx_sz += api->report(idx);
     }
 
+    if (FIELD_IS(fs, INDEXFLD_T_VECTOR)) {
+      VecSimParams vec_params = fs->vectorOpts.vecSimParams;
+      VecSimAlgo algo = vec_params.algo;
+
+      if (algo == VecSimAlgo_TIERED) {
+        VecSimParams *backend_params = vec_params.algoParams.tieredParams.primaryIndexParams;
```
Comment (alonre24) [inline src/info_command.c:165]: You can save some code if you use:
```
if (algo == VecSimAlgo_TIERED) {
        params = *vec_params.algoParams.tieredParams.primaryIndexParams;
}
```
then reply for the algorithm, and then continue with the rest as you did (separate by algo)

**Review by alonre24 — COMMENTED** [review]
Nice!!

File: tests/pytests/test_info.py (hunk 2 above)
Comment (meiravgri) [reply tests/pytests/test_info.py:77]: right click + format

**Review by alonre24 — DISMISSED** [review]

**Review by alonre24 — APPROVED** [review]

**Pull request merged**